### ISSUE TITLE
Upgrade from rancher 2.5.13 to 2.5.14 fails...

### ISSUE THREAD

**Issue by seimic** [rancher/rancher#37859]
**Rancher Server Setup**
- Rancher version:
 Currently 2.5.13

- Installation option (Docker install/Helm Chart):
   RKE2 (rancherd)

**Information about the Cluster**
- Kubernetes version:
   v1.20.7+rke2r2
- Cluster Type (Local/Downstream):
   Custom (Hetzner):
   
**Describe the bug**

Upgrade fails. "docker.io/rancher/rancher:###-dirty" doesn't exist. I was expecting it to pull rancher/**rancher:v2.5.14**

**To Reproduce**

```sh
curl -sfL https://get.rancher.io | sudo INSTALL_RANCHERD_VERSION=v2.5.14 sh -
sudo systemctl enable rancherd-server
sudo systemctl stop rancherd-server
sudo systemctl start rancherd-server
```

**Result**

Upgrade fails. docker.io/rancher/rancher:###-dirty (what's dirty here? :) ) doesn't exist.
After this, back to v2.5.13 without issues.

**Expected Result**
Upgrade to v2.5.14

**Screenshots**

![image](https://user-images.githubusercontent.com/34323525/170999104-a419b0df-7a55-49b1-932a-0e66d5ef40b4.png)
![image](https://user-images.githubusercontent.com/34323525/170999210-0d5e7380-d470-4cf5-8db4-246e1c4375cb.png)

**Comment by hans99** [comment]
Same problem here, looks like the app specifies this dirty-build version and not v2.5.14 as it should.

**Comment by cbron** [comment]
This was also a problem in 2.5.15 as it was essentially a clone of 2.5.14, but will be fixed for 2.5.16. 

To QA this, an RC should show a clean version, but we should also confirm that 2.5.15-rc1 did show the "dirty" tag.

**Comment by brudnak** [comment]
# :white_check_mark: PASSED

## Reproduction Environment

| Component                 | Version / Type         |
| ------------------------- | ---------------------- |
| Rancher version           | 2.5.15-rc1             |
| Installation option       | helm high availability |
| RKE binary version used   | 1.20.20                |
| If Helm Chart k8s cluster | v1.20.15               |
| Cert Details              | let's encrypt          |
| Helm version              | v2.16.8-rancher1       |
| Downstream cluster type   | rke1 linode            |
| Downstream K8s version    | v1.20.15               |
| Logged in user role       | administrator          |
| Browser type              | google chrome          |
| Browser version           | 103.0.5060.114         |

**ADDITIONAL SETUP**

- Helm commands for Rancher installation:

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$URL_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=nginx \
  --version 2.5.15-rc1
```

## Reproduction steps

1. Create a downstream rke1 linode cluster
2. Wait for the downstream linode cluster to finish provisioning
3. Check the logs of the `cattle-cluster-agent` on the downstream

```log
time="<DATE_TIME_REDACTED>" level=info msg="Listening on /tmp/log.sock"
time="<DATE_TIME_REDACTED>" level=info msg="Rancher agent version 27f52ed08-dirty is starting"
```
  
## Additional Info

## RESULTS

#### :white_check_mark: Expected
Expected logs **without** "dirty"

#### :x: Actual
Getting dirty logs

<hr>
  
## Validation Environment

| Component                 | Version / Type                                                                     |
| ------------------------- | ---------------------------------------------------------------------------------- |
| Rancher version           | v2.5-4b56580160fe623dd84f18b3a6d46a0f78a12f44-head                                 |
| Rancher commit link       | https://github.com/rancher/rancher/commit/4b56580160fe623dd84f18b3a6d46a0f78a12f44 |
| Installation option       | helm high availability                                                             |
| RKE binary version used   | 1.20.20                                                                            |
| If Helm Chart k8s cluster | v1.20.15                                                                           |
| Cert Details              | let's encrypt                                                                      |
| Helm version              | v2.16.8-rancher1                                                                   |
| Downstream cluster type   | rke1 linode                                                                        |
| Downstream K8s version    | v1.20.15                                                                           |
| Logged in user role       | administrator                                                                      |
| Browser type              | google chrome                                                                      |
| Browser version           | 103.0.5060.114                                                                     |

**ADDITIONAL SETUP**

- Helm commands for Rancher installation:

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$URL_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=nginx \
  --set rancherImageTag=v2.5-head
```

## Validation steps

1. Create a downstream rke1 linode cluster
2. Wait for the downstream linode cluster to finish provisioning
3. Check the logs of the `cattle-cluster-agent` on the downstream
4. There is no occurrence of **"dirty"** in the logs
  
## Additional Info

## RESULTS

#### :white_check_mark: Expected
No occurrence of **"dirty"** in the logs

#### :white_check_mark:Actual
No occurrence of **"dirty"** in the logs
